net/example_test.go

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package xnet_test
+
+import (
+	"io"
+	"log"
+	"net"
+
+	"github.com/m3db/m3x/net"
+	"github.com/m3db/m3x/retry"
+)
+
+func ExampleStartForeverAcceptLoop() {
+	// Create a new listener.
+	l, err := net.Listen("tcp", ":0")
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer l.Close()
+
+	// Start accepting incoming connections.
+	var (
+		opts          = xretry.NewOptions()
+		connCh, errCh = xnet.StartAcceptLoop(l, opts)
+	)
+
+	for {
```

Comment:
same as above: does this run in the test suite?

Reply:
I think in this case we want to just compile the test since otherwise we would have to introduce some kind of timeout to close the server which may make the example less clear.

Reply:
Sounds good